### PR TITLE
Try to make docker credentials experience a bit better

### LINKED ISSUE #1413: Docker-compose broken after `neuro config docker`
#### Description:
`neuro config docker` breaks unrelated docker-compose setups

#### Example:
Steps to reproduce the behaviour:
* Create any docker-compose file and try to run it after `neuro logout`

#### Expected behavior:
At least my unrelated docker-compose file is run, maybe with some warning, but not error.

#### Additional context:
Prints error as follows:
```
⇒  make test-integration
docker-compose -f tests/integration/docker-compose.yaml pull -q --ignore-pull-failures
docker-compose -f tests/integration/docker-compose.yaml up -d --build
Building runner
Traceback (most recent call last):
  File "/Users/taras/.pyenv/versions/3.7.4/bin/docker-credential-neuro", line 11, in <module>
    load_entry_point('neuromation', 'console_scripts', 'docker-credential-neuro')()
  File "/Users/taras/workspace/platform-client-python/neuromation/cli/docker_credential_helper.py", line 45, in main
    run(async_main(action))
  File "/Users/taras/workspace/platform-client-python/neuromation/cli/asyncio_utils.py", line 116, in run
    return runner.run(main)
  File "/Users/taras/workspace/platform-client-python/neuromation/cli/asyncio_utils.py", line 48, in run
    return self._loop.run_until_complete(main_task)
  File "/Users/taras/.pyenv/versions/3.7.4/lib/python3.7/asyncio/base_events.py", line 579, in run_until_complete
    return future.result()
  File "/Users/taras/workspace/platform-client-python/neuromation/cli/docker_credential_helper.py", line 21, in async_main
    async with get() as client:
  File "/Users/taras/workspace/platform-client-python/neuromation/api/utils.py", line 68, in __aenter__
    self._ret = await self._coro
  File "/Users/taras/workspace/platform-client-python/neuromation/api/__init__.py", line 125, in _get
    return await Factory(path).get(timeout=timeout)
  File "/Users/taras/workspace/platform-client-python/neuromation/api/config_factory.py", line 72, in get
    client = Client._create(session, self._path, self._trace_id)
  File "/Users/taras/workspace/platform-client-python/neuromation/api/utils.py", line 53, in _create
    return super().__call__(*args, **kwargs)
  File "/Users/taras/workspace/platform-client-python/neuromation/api/client.py", line 33, in __init__
    self._config._load()
  File "/Users/taras/workspace/platform-client-python/neuromation/api/config.py", line 69, in _load
    ret = self.__config_data = _load(self._path)
  File "/Users/taras/workspace/platform-client-python/neuromation/api/config.py", line 279, in _load
    with _open_db_ro(path) as db:
  File "/Users/taras/.pyenv/versions/3.7.4/lib/python3.7/contextlib.py", line 112, in __enter__
    return next(self.gen)
  File "/Users/taras/workspace/platform-client-python/neuromation/api/config.py", line 243, in _open_db_ro
    raise ConfigError(f"Config at {path} does not exists. Please login.")
neuromation.api.config.ConfigError: Config at /Users/taras/.neuro does not exists. Please login.
Traceback (most recent call last):
  File "site-packages/dockerpycreds/store.py", line 80, in _execute
  File "subprocess.py", line 356, in check_output
  File "subprocess.py", line 438, in run
subprocess.CalledProcessError: Command '['/Users/taras/.pyenv/shims/docker-credential-neuro', 'get']' returned non-zero exit status 1.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "site-packages/docker/auth.py", line 264, in _resolve_authconfig_credstore
  File "site-packages/dockerpycreds/store.py", line 35, in get
  File "site-packages/dockerpycreds/store.py", line 93, in _execute
dockerpycreds.errors.StoreError: Credentials store docker-credential-neuro exited with "".

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "docker-compose", line 6, in <module>
  File "compose/cli/main.py", line 71, in main
  File "compose/cli/main.py", line 127, in perform_command
  File "compose/cli/main.py", line 1085, in up
  File "compose/cli/main.py", line 1081, in up
  File "compose/project.py", line 527, in up
  File "compose/service.py", line 344, in ensure_image_exists
  File "compose/service.py", line 1084, in build
  File "site-packages/docker/api/build.py", line 260, in build
  File "site-packages/docker/api/build.py", line 307, in _set_auth_headers
  File "site-packages/docker/auth.py", line 310, in get_all_credentials
  File "site-packages/docker/auth.py", line 281, in _resolve_authconfig_credstore
docker.errors.DockerException: Credentials store error: StoreError('Credentials store docker-credential-neuro exited with "".',)
[73174] Failed to execute script docker-compose
```

### PR DESCRIPTION
**Pull request by tvoinarovskyi**
Fixes #1413

### CODE REVIEW

**Comment by tvoinarovskyi** [comment]
For docker-compose case it looks something like:
```
```

**Comment by tvoinarovskyi** [comment]
@asvetlov Added

**Comment by tvoinarovskyi** [comment]
Docker pull error now looks like this:
```
⇒  docker pull registry-dev.neu.ro/tvoinarovskyi/e2e-banana-image
Using default tag: latest
Could not resolve correct credentials for registry-dev.neu.ro. Please re-login using `neuro login` or switch to the related cluster.
Error response from daemon: unauthorized: authentication required
```

**Comment by tvoinarovskyi** [comment]
Docker login is a bit worse than it was:
```
taras@Tarass-MacBook-Pro:~/workspace/platform-client-python|fix_docker_credentials⚡
⇒  docker login registry-dev.neu.ro
Could not resolve correct credentials for registry-dev.neu.ro. Please re-login using `neuro login` or switch to the related cluster.
Authenticating with existing credentials...
Login did not succeed, error: Error response from daemon: login attempt to https://registry-dev.neu.ro/v2/ failed with status: 401 Unauthorized
Username (invalid):
Password:
Error: Password Required
```

File: neuromation/cli/docker_credential_helper.py [hunk 1]
```diff
@@ -18,19 +18,29 @@ def error(message: str, exit_code: int) -> None:
...
-                )
-            token = await config.token()
-            payload = {"Username": "token", "Secret": token}
-            print(dumps(payload))
+        registry = sys.stdin.readline().strip()
+        try:
+            async with get() as client:
+                config = client.config
+                neuro_registry = config.registry_url.host
+                if registry != neuro_registry:
```
Comment (dalazx) [inline neuromation/cli/docker_credential_helper.py]: seems like here we need to check against 
https://github.com/neuromation/platform-client-python/blob/b1b5df046c0c162f305435180f21fdd38a38fba1/neuromation/api/config.py#L89-L91

Comment (dalazx) [reply neuromation/cli/docker_credential_helper.py]: here 
https://github.com/neuromation/platform-client-python/blob/b1b5df046c0c162f305435180f21fdd38a38fba1/neuromation/cli/config.py#L193-L194
we are adding a single cluster, although a user can have multiple clusters.

**Pull request closed without merging**